internal/output/table.go

```diff
@@ -207,6 +209,11 @@ func tableBuilderInner(vulnResult *models.VulnerabilityResults, calledVulns bool
 				source.Path = sourcePath
 			}
 
+			// Ensure that groups are sorted consistently using the first ID in each group
+			slices.SortFunc(pkg.Groups, func(a, b models.GroupInfo) int {
```

Comment:
note that this is actually modifying the value of `pkg.Groups` on the struct, which applies outside of this function - I actually thought Go didn't work like this, but apparently it does.

I have somewhat mixed feelings about doing it, as generally this is considered bad practice (especially as it's not obvious), but it saves us a few lines of code + a variable, and any bug that it causes I would expect to be because whatever is being impacted should itself be doing a sort if it cares about the order...

Reply:
I think the best approach here is probably to replace everything with [Output/Result](https://github.com/google/osv-scanner/blob/5f2f7354b95c895ac7749696c80eec7fed62dab1/internal/output/output_result.go#L20), which is already sorted and contains all the data we need to display (even extras). I should probably update it for consistency (since container scanning result uses Output/Result).

Reply:
Should this sorting be done for the VulnerabilityResults rather than in the output?